### PR TITLE
Adding initial loging for utilizing a platform identifier CH-19944

### PR DESCRIPTION
**Pull request by CTDoriott**

### CODE REVIEW

File: src/Http/Client.php [hunk 1]
```diff
@@ -217,12 +229,17 @@ public function post(
...
-        if (! SecurityClient::validateAuthUser($authUsername)) {
+        if (! SecurityClient::validateAuthUser((string) $authUsername)) {
             throw new HttpException(
                 sprintf('An auth username is required for NS8 %s requests.', self::POST_REQUEST_TYPE)
             );
         }
 
+        $platformIdentifier = $this->getPlatformIdentifier();
+        if (! empty($platformIdentifier)) {
+            $data['platform_uuid'] = $platformIdentifier;
```
Comment (CTDoriott) [inline src/Http/Client.php]: @crfroehlich This is currently what would be passed in with all requests including events & actions to represent the unique platform identifier. Please let me know if you'd prefer a different name or naming convention.

File: src/Http/Client.php [hunk 2]
```diff
@@ -445,6 +462,30 @@ public function setAuthUsername(string $authUsername) : IProtectClient
...
+     */
+    public function getPlatformIdentifier() : ?string
+    {
+        return $this->platformIdentifier;
+    }
+
+    /**
+     * Set Platform identifier used in post requests
+     *
+     * @param srtring $platformIdentifier Platform identifier to use when sending requests
```
Comment (davehensleyns8) [inline src/Http/Client.php]: Typo

File: tests/Actions/ClientTest.php [hunk 3]
```diff
@@ -68,6 +68,7 @@ class ClientTest extends TestCase
      * @covers NS8\ProtectSDK\Http\Client::setAccessToken
      * @covers NS8\ProtectSDK\Http\Client::setAuthUsername
      * @covers NS8\ProtectSDK\Http\Client::setSessionData
+     * @covers NS8\ProtectSDK\Http\Client::getPlatformIdentifier
```
Comment (davehensleyns8) [inline tests/Actions/ClientTest.php]: I know it's silly but I hate it when things fall out of alphabetical order.

File: tests/Actions/ClientTest.php [hunk 4]
```diff
@@ -120,6 +121,7 @@ public function testSetAction() : void
...
```
Comment (davehensleyns8) [inline tests/Actions/ClientTest.php]: Same thing here.

File: tests/Analytics/ClientTest.php [hunk 5]
```diff
@@ -35,6 +35,7 @@ class ClientTest extends TestCase
      * @covers NS8\ProtectSDK\Config\Manager::setValue
      * @covers NS8\ProtectSDK\Config\Manager::setValueWithoutValidation
      * @covers NS8\ProtectSDK\Config\Manager::validateKeyCanChange
+     * @covers NS8\ProtectSDK\Config\Manager::doesValueExist
```
Comment (davehensleyns8) [inline tests/Analytics/ClientTest.php]: And here.

File: tests/Analytics/ClientTest.php [hunk 6]
```diff
@@ -58,6 +59,7 @@ public function testGetTrueStatsUrl() : void
...
```
Comment (davehensleyns8) [inline tests/Analytics/ClientTest.php]: ...and all the rest of them...

File: public/en/platform/protect-sdk-php/configuration-documentation.md [hunk 7]
```diff
@@ -84,3 +84,5 @@ use NS8\ProtectSDK\Config\Manager as ConfigManager;
 // Does Foo config value exist?
 $doesFooExist = ConfigManager::doesValueExist('foo');
 ```
+
+
```
Comment (davehensleyns8) [inline public/en/platform/protect-sdk-php/configuration-documentation.md]: This doesn't seem super necessary but I guess there's no harm in a little extra white space.

**Pull request merged**